nodejs/webrtc-hello-again/server.js

```diff
@@ -0,0 +1,257 @@
+      console.log("Add PSTN caller to the session")
+      webRTCController.addParticipantToSession(accountId, getSessionId(), createResponse.participant.id, body, 
+        function(error, response, context) {
+          // console.log("addParticipantToSession response:", response)         
+      }).catch(error => {
+        console.error('Error in addParticipantToSession:', error);
+        res.send('{"status":"Failed to add PSTN call to session"}')
+      });
+
+      var body = new BandwidthVoice.ApiCreateCallRequest({
+        "from": process.env.FROM_NUMBER,
+        "to": process.env.OUTBOUND_PHONE_NUMBER,
+        "applicationId": process.env.VOICE_APPLICATION_ID,
+        "answerUrl": process.env.BASE_CALLBACK_URL + 'callAnswered',
+        "answerMethod": "POST",
+        "callTimeout": 10
+      });
+
+      console.log("start the call to", process.env.OUTBOUND_PHONE_NUMBER)
+      try{
+        var response = await voiceController.createCall(accountId, body)
+        callId = response.callId;
+        calls.set(response.callId, participant_response)
+        // console.log("voice API reponse: ", response);      
+        res.send('{"status":"ringing"}')
+  
+      } catch (e) {
+        console.error(`error calling ${process.env.OUTBOUND_PHONE_NUMBER}`, e);
+
+        res.send('{"status":"call failed"}')
+      }
+  }).catch(error => {
+    console.error('Error in createParticipant:', error);
+    res.send('{"status":"Failed to get token"}')
+  });  
+
+
+
+});
+
+/**
+ * Bandwidth's Voice API will hit this endpoint when an outgoing call is answered
+ */
+app.post("/callAnswered", async (req, res) => {
+  // console.log("Callback from Voice system:", req.body)
+  const callId = req.body.callId;
+  console.log(`received answered callback for call ${callId} tp ${req.body.to}`);
+
+  const participant = calls.get(callId);
+  if (!participant) {
+    console.log(`no participant found for ${callId}!`);
+    res.status(400).send();
+    return;
+  }
+
+  // This is the response payload that we will send back to the Voice API to transfer the call into the WebRTC session
+  // const bxml = webRTCController.generateTransferBxml(participant.token);
+  const bxml = generateTransferBxml(participant.token);
+  console.log(`transferring call ${callId} to session ${sessionId}`)
+
+  // Send the payload back to the Voice API
+  res.contentType("application/xml").send(bxml);
+  console.log("transferred")
+});
+
+/** 
+ * End the Phone Call
+ */
+app.get('/endPSTNCall', async (req, res) => {
+  if (getSessionId() == false || callId == false){
+    console.log("No session found; can't end PSTN call")
+    res.send('{"status":"Failed to end PSTN call; no session id found"}')
+    return;
+  }
+
+  var body = new BandwidthVoice.ApiModifyCallRequest({"state": "completed"});
+  try{
+    var response = await voiceController.modifyCall(accountId, callId, body)
+    res.send('{"status":"hungup"}')
```

Comment:
Not to get too hungup on this... oh nevermind...

Reply:
🤣